src/hooks/useENS.ts

```diff
@@ -1,15 +1,17 @@
 import { ethers } from "ethers";
 import { useEffect, useState } from "react";
 import { useWeb3Context } from "./web3Context";
+import { NetworkId } from "../constants";
 
 const useENS = (address: string) => {
-  const { provider } = useWeb3Context();
+  const { provider, networkId, providerInitialized } = useWeb3Context();
   const [ensName, setENSName] = useState<string | null>(null);
   const [ensAvatar, setENSAvatar] = useState<string | null>(null);
+  const ensSupport: boolean = networkId === NetworkId.MAINNET || networkId === NetworkId.TESTNET_RINKEBY;
 
   useEffect(() => {
     const resolveENS = async () => {
-      if (ethers.utils.isAddress(address)) {
+      if (providerInitialized && ensSupport && ethers.utils.isAddress(address)) {
         try {
           let ensName = await provider.lookupAddress(address);
           let avatar = ensName ? await provider.getAvatar(ensName) : null;
```

Comment:
I'm not sure i want to hard code in an if statement when the error is already caught on line 13/15. @0xJem wdyt

Reply:
I personally would prefer to not spew irrelevant/avoidable errors into the console.

In terms of code style, I would suggest taking an early exit approach.
```
if (!providerInitialized) return;

if (!ensSupport) return;

if (!ethers.utils.isAddress(address)) return;
```

These could be combined into one line too, if one prefers.

Reply:
It's not the styling of the if that I don't like. 
In general the `try/catch` is automatic and handles all cases. An if statement is manual & can only handle what you manually maintain.

Since the root issue you're pointing out is the logged error you should revise how the error is logged rather than adding handling outside the error.

Irrelevant to this particular PR but as further context: This is the same approach I requested for Give. Add better handling within the error to make it more useful. That would be another good place you guys could go to cleanup.

Reply:
Is conditional error logging acceptable in this case? Only unexpected/unforeseen errors would then print to console.

Ofc, long term, a more robust error handling setup would be created

Reply:
Yes, I'm saying we need better error handling in general.

I would say within the catch we don't log the error we just log `ENS not supported on ChainId: ${NetworkId}`... but we're bike-shedding at this point. It's not important enough either way.

If you guys want to hard code in an ENS Support check for one chain that's fine for now.